packages/contracts/contracts/discovery/Controller.sol

```diff
@@ -34,8 +39,16 @@ contract Controller is IController {
         return batches[id];
     }
 
-    function approveProjectByOwner(uint256 id) external {
-        revert("not implemented");
+    function approveProjectByOwner(uint256 id)
+        external
+        onlyRole(DEFAULT_ADMIN_ROLE)
+    {
+        require(
+            projects[id].status == ProjectStatus.Created,
+            "invalid state for project"
+        );
+
+        projects[id].status = ProjectStatus.Whitelisted;
```

Comment:
I don't think this is gonna hold in the future.
a project will have my criteria to fulfill to be whitelisted (e.g.: owner whitelist, legal whitelist, token supply already sent, all data filled in).
So you can't have this function migrate directly to that state
probably we need to have multiple boolean fields on the project, and keep it in a `Created` state